internal/webhooks/machine.go

```diff
@@ -83,6 +84,10 @@ func (webhook *Machine) Default(_ context.Context, obj runtime.Object) error {
 		m.Spec.NodeDeletionTimeout = &metav1.Duration{Duration: defaultNodeDeletionTimeout}
 	}
 
+	if m.Spec.MinReadySeconds == nil {
```

Comment:
Q: can we drop this now that we have // +default=0 on the field definition?

Reply:
We need to keep it because the +default=0 isn't used.

Reply:
We should delete the defaulting so we preserve user intent (same for other webhooks).

Controller already treats nil the same as 0

(same for other webhooks)

Reply:
that's correct, we want to drop this, see https://github.com/kubernetes-sigs/cluster-api/pull/12153#discussion_r2084056184